sos/collector/clusters/openstack.py

```diff
@@ -0,0 +1,68 @@
+# Copyright Red Hat 2022, Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import yaml
```

Comment:
Some tests are failing with `Could not initialize 'collect': No module named 'yaml'`. The module is provided by `python3-pyyaml` which is a requirement of `python3-magic` we (newly) need for file type detection. So how the module could be missed?

Reply:
ACK to the PR otherwise, no idea why the SCI tests could fail this way.

Reply:
I've been looking into this a bit more.

There is a possibility of using something like `openstack overcloud profiles list` and then a follow up `openstack` command that maps the node names to addresses - however the architecture of OSP in regards to where this command is functional is not immediately clear to me.

I've also clarified with the support team that this ansible inventory file is updated as the environment changes, and is the defacto source of truth.

I'm a bit torn here. One the one hand I am _really_ not fond of adding _another_ dependency, but this approach to enumeration is a lot more straight forward and reliable than relying on a chain of commands that output a table and having to grok that.